### PR TITLE
conf parser: add config_parse_ip_port

### PR DESCRIPTION
**Pull request by ssahani**

### CODE REVIEW

File: src/shared/conf-parser.c [hunk 1]
```diff
@@ -960,3 +960,38 @@ int config_parse_ifname(
...
+        uint16_t port;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        if (isempty(rvalue))
+                return 0;
```
Comment (poettering) [inline src/shared/conf-parser.c]: hmm, i wonder what the best approach here would be. Normally we say that assigning the empty string to a setting this should have the effect of resetting things to the "default", for whatever that might be. Now, implementing this properly here is difficult, since we don't know the right default, as it might be different for each user of this function. Hoewver, I think it would make sense to assign the port "0" in this case, and then make sure that the code using this setting treats 0 as "the default" when using the field.

Long story short: please change this to: 
```c
if (isempty(rvalue)) {
        *s = 0;
        return 0;
}
```

**Comment by poettering** [comment]
Hmm, does VXLAN/GENEVE have any sensible "default" port? i.e. something that is usually used? If so, I figure the networkd code should fall back to that default port in each case if the field is found to be initialized to 0. Also see the other comments above.

If there's no accepted "default" port, maybe we should just make some up, and document it.

**Comment by ssahani** [comment]
```
sudo ip link add vxlan0 type vxlan id 42 group 239.1.1.1 dev enp0s25
vxlan: destination port not specified
Will use Linux kernel default (non-standard value)
Use 'dstport 4789' to get the IANA assigned value
Use 'dstport 0' to get default and quiet this message
```
yes if we supply `0` to kernel the the default values are used. I have updated the doc.

File: man/systemd.netdev.xml [hunk 2]
```diff
@@ -600,8 +600,8 @@
       <listitem>
         <para>Configures the default destination UDP port on a per-device basis.
         If destination port is not specified then Linux kernel default will be used.
-        Set destination port 4789 to get the IANA assigned value,
-        and destination port 0 to get default values.</para>
+        Set destination port 4789 to get the IANA assigned value. If destination
+        port is set to 0 then default value (4789) is used.</para>
```
Comment (poettering) [inline man/systemd.netdev.xml]: Nah, this is not correct: if you specify the port as "0", this should result in a parse failure now, and rightfully so. However, if you assign the empty string, then you get the IANA default value. These two sentences should hence be replaced by something like this:

```If not set or If the destination port is assigned the empty string the default port of 4789 is used.```

File: man/systemd.netdev.xml [hunk 3]
```diff
@@ -676,7 +676,8 @@
     <varlistentry>
       <term><varname>DestinationPort=</varname></term>
       <listitem>
-        <para>Specifies destination port. Defaults to 6081.</para>
+        <para>Specifies destination port. Defaults to 6081.If destination
+        port is set to 0 then default value (6081) is used.</para>
```
Comment (poettering) [inline man/systemd.netdev.xml]: Similar here. Replace the two sentences by:

```If not set or assigned the empty string, the default port of 6081 is used.```
